### ISSUE TITLE
Observable errors from websocket `@SubscribeMessage` are not caught by exception filters, causing server crash

### ISSUE THREAD

**Issue by zakhenry** [nestjs/nest#1285]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

If I return an observable that errors from within a `@WebsocketGateway` method that has `@SubscribeMessage`, the server crashes rather than invoking an error handler.

## Expected behavior

Either the global error handler should catch the error and emit it to the client, or if set, the custom WsException.catch method should be invoked with the error.

## Minimal reproduction of the problem with instructions

```ts
import { Injectable, UseFilters } from '@nestjs/common';
import { OnGatewayInit, SubscribeMessage, WebSocketGateway, WebSocketServer, WsResponse } from '@nestjs/websockets';
import { Observable, throwError } from 'rxjs';
import { map } from 'rxjs/operators';
import { Namespace, Socket } from 'socket.io';

@WebSocketGateway()
export class SocketGateway {

  @SubscribeMessage('ping')
  public pong(
    client: Socket,
    data: any,
  ): Observable<WsResponse<any>> {
    
    return throwError(); // this causes server crash, not a handled exception
  }
}
```

## Environment

<pre><code>
Nest version: 5.3.0

**Comment by kamilmysliwiec** [comment]
Thanks for reporting! Fixed by #1346

**Comment by zakhenry** [comment]
@kamilmysliwiec excellent thanks! Sorry it was a bit of a tough one to give a repro for

**Comment by kamilmysliwiec** [comment]
Fixed in 5.5.0 :)
